src/tools/ts-generator/linksTSgenerator.cpp

```diff
@@ -0,0 +1,346 @@
+        if (fs::exists(modulationFileDirect))
+        {
+            to_return = link.modulationCapacityDirect.loadFromCSVFile(
+                    modulationFileDirect.string(),
+                    1,
+                    HOURS_PER_YEAR)
+                        && to_return;
+        }
+
+        auto modulationFileIndirect = preproFileRoot;
+        modulationFileIndirect += "_mod_indirect.txt";
+        if (fs::exists(modulationFileIndirect))
+        {
+            to_return = link.modulationCapacityIndirect.loadFromCSVFile(
+                    modulationFileIndirect.string(),
+                    1,
+                    HOURS_PER_YEAR)
+                        && to_return;
+        }
+        // Makes it possible to skip a link's TS generation when time comes
+        link.hasValidData = link.hasValidData && to_return;
+        return to_return;
+    }
+
+void readPreproTimeSeries(std::vector<LinkTSgenerationParams>& linkList,
+                          fs::path toLinksDir)
+{
+    for(auto& link : linkList)
+    {
+        std::string sourceAreaName = link.namesPair.first;
+        fs::path sourceAreaDir = toLinksDir / sourceAreaName;
+        if (! readLinkPreproTimeSeries(link, sourceAreaDir))
+        {
+            logs.warning() << "Could not load all prepro data for link '"
+                           << link.namesPair.first << "." << link.namesPair.second << "'";
+        }
+    }
+}
+
+std::string DateAndTime()
+{
+    YString to_return;
+    unsigned int now = Yuni::DateTime::Now();
+    Yuni::DateTime::TimestampToString(to_return, "%Y%m%d-%H%M", now);
+    return to_return.to<std::string>();
+}
+
+// ==================
+// Class methods
+// ==================
+LinksTSgenerator::LinksTSgenerator(Settings & settings) :
+    studyFolder_(settings.studyFolder),
+    linksFromCmdLineOptions_(settings.linksListToGen),
+    generateTSforAllLinks_(settings.allLinks)
+{
+}
+
+void LinksTSgenerator::extractData()
+{
+    auto allLinksPairs = extractLinkNamesFromStudy();
+
+    LinkPairs namesLinksToGenerate;
+    if (generateTSforAllLinks_)
+        namesLinksToGenerate = allLinksPairs;
+    else
+        namesLinksToGenerate = extractLinkNamesFromCmdLine(allLinksPairs);
+
+    linkList_ = CreateLinkList(namesLinksToGenerate);
+    extractLinksSpecificTSparameters();
+
+    generalParams_ = readGeneralParamsForLinksTS();
+}
+
+LinkPairs LinksTSgenerator::extractLinkNamesFromStudy()
+{
+    LinkPairs to_return;
+    fs::path linksDir = studyFolder_ / "input" / "links";
+    for (auto const& item : fs::directory_iterator{linksDir})
+    {
+        if (item.is_directory())
+        {
+            std::string sourceAreaName = item.path().filename().generic_string();
+            auto targetAreas = extractTargetAreas(item);
+            for (auto& targetAreaName : targetAreas)
+            {
+                auto linkPair = std::make_pair(sourceAreaName, targetAreaName);
+                to_return.push_back(linkPair);
```

Comment:
std::transform?

Reply:
I think it's clear enough and short enough as it is.